From 8fcec302bfc87e056fd2e761c3c88c4bf5cc1785 Mon Sep 17 00:00:00 2001
From: Xu Xuehui <xxh@rock-chips.com>
Date: Wed, 21 Jun 2017 17:14:33 +0800
Subject: [PATCH] net: wireless: rockchip_wlan: fix bug for ap6335

1. fix when start softap, wifi driver crash
2. fix p2p mac addr not correct, this case p2p can not use

Change-Id: Ia9b5d7ac660ec9b954a682345c52778f7534229e
Signed-off-by: Xu Xuehui <xxh@rock-chips.com>
---
 drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/Makefile     | 2 +-
 drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_config.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/Makefile b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/Makefile
index a8131575a4e7..a6a4be363d68 100644
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/Makefile
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/Makefile
@@ -68,7 +68,7 @@ bcmdhd-objs += $(DHDOFILES)
 
 #ifeq ($(CONFIG_MACH_PLATFORM),y)
 DHDOFILES += dhd_gpio.o
-DHDCFLAGS += -DCUSTOMER_HW -DDHD_OF_SUPPORT -DGET_CUSTOM_MAC_ENABLE
+DHDCFLAGS += -DCUSTOMER_HW -DDHD_OF_SUPPORT
 #DHDCFLAGS += -DBCMWAPI_WPI -DBCMWAPI_WAI
 #endif
 
diff --git a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_config.c b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_config.c
index 3c5c9f29a497..437efb736bbd 100755
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_config.c
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/dhd_config.c
@@ -346,7 +346,7 @@ dhd_conf_set_fw_name_by_chip(dhd_pub_t *dhd, char *fw_path, char *nv_path)
 	int fw_type, ag_type;
 	uint chip, chiprev;
 	int i,j;
-	char fw_tail[10];
+	char fw_tail[20];
 
 	chip = dhd->conf->chip;
 	chiprev = dhd->conf->chiprev;
-- 
2.35.3

